### PR TITLE
Continuous deployment for updating metrics

### PR DESCRIPTION
**Pull request by bendnorman**
This PR enables continuous deployment of PUDL usage metrics. Main changes:
- Created a daily github action called `load-metrics` that runs the `run_data_update.py` script. 
- Added some logic in `respositories.py` to collect all data source jobs. These jobs are organized into different repositories based on the database manager they use. 
- Create a new [dagster resource](https://docs.dagster.io/concepts/resources) for Postgres databases so jobs can load data to a Google Cloud SQL Postgres db. 
- Skip downstream ops if there are no datasette logs to process. 
- Improved documentation.

### CODE REVIEW

File: src/usage_metrics/jobs/datasette.py [hunk 1]
```diff
@@ -39,13 +24,32 @@ def datasette_weekly_partition(start: datetime, end: datetime):
...
+def transform(raw_logs: pd.DataFrame) -> pd.DataFrame:
+    """Transform datasette logs."""
+    df = da.unpack_httprequests(raw_logs)
+    df = da.parse_urls(df)
+    return da.geocode_ips(df)
+
+
+@graph
+def process_datasette_logs():
+    """Process datasette logs locally using a SQLite database."""
```
Comment (zschira) [inline src/usage_metrics/jobs/datasette.py:37]: I believe this comment should be updated to remove locally

File: src/usage_metrics/resources/postgres.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        self,
+        user: str,
+        password: str,
+        db: str,
+        ip: str,
+        port: str,
+        clobber: bool = False,
+    ) -> None:
+        """
+        Initialize SQLiteManager object.
```
Comment (zschira) [inline src/usage_metrics/resources/postgres.py:24]: PostgresManager

**Review by zschira — COMMENTED** [review]
Looks great! Just a few minor things I found

File: src/usage_metrics/repository.py [hunk 3]
```diff
@@ -1,10 +1,55 @@
...
+from usage_metrics.resources.postgres import postgres_manager
+from usage_metrics.resources.sqlite import sqlite_manager
+
+gcp_jobs = []
+local_jobs = []
+
+jobs_subpackage_path = Path(__file__).parent / "jobs"
+modules = [str(jobs_subpackage_path)]
+
+for module_info in pkgutil.iter_modules(modules):
```
[the remark this answers is not in the document]
Comment (bendnorman) [reply src/usage_metrics/repository.py:28]: Hmm good question. It seems common to create a local and production job from graphs but I haven't seen anything about creating separate repositories for each type of job. Dagster doesn't really prescribe patterns for using repositories:
> A repository is a collection of software-defined assets, jobs, schedules, and sensors. Repositories are loaded as a unit by the Dagster CLI, Dagit and the Dagster Daemon. A repository is a convenient way to organize your job and other definitions. Each repository:

File: src/usage_metrics/jobs/datasette.py (hunk 1 above)
Comment (bendnorman) [reply src/usage_metrics/jobs/datasette.py]: Ah thanks

File: src/usage_metrics/resources/postgres.py (hunk 2 above)
Comment (bendnorman) [reply src/usage_metrics/resources/postgres.py]: Fixed

**Pull request merged**